src/realm/sync/noinst/client_reset.cpp

```diff
@@ -357,11 +379,11 @@ void transfer_group(const Transaction& group_src, Transaction& group_dst, util::
 
         for (const Obj& src : *table_src) {
             auto src_pk = src.get_primary_key();
-            bool updated = false;
             // get or create the object
-            auto dst = table_dst->create_object_with_primary_key(src_pk, &updated);
+            auto dst = table_dst->get_object_with_primary_key(src_pk);
```

Comment:
~~This will create the object if it doesn't exist?~~
Or it should have already been created by your changes above?

Reply:
Yup. I updated the comment above.